system/smbus.c

```diff
@@ -182,14 +182,38 @@ static void print_spdi(spd_info spdi, uint8_t lidx)
     }
 }
 
+static void read_sku(char *sku, uint8_t slot_idx, uint16_t offset, uint8_t max_len)
+{
+    uint8_t sku_len;
+
+    if (max_len > SPD_SKU_LEN)
+        max_len = SPD_SKU_LEN;
+
+    for (sku_len = 0; sku_len < max_len; sku_len++) {
+        uint8_t sku_byte = get_spd(slot_idx, offset + sku_len);
+
+        // Stop on the first non-ASCII char
+        if (sku_byte < 0x20 || sku_byte > 0x7F)
+            break;
+
+        sku[sku_len] = (char)sku_byte;
+    }
+
+    // Trim spaces from the end
+    while (sku_len && sku[sku_len - 1] == 0x20)
```

Comment:
Many DRAM manufacturers do not follow the JEDEC SPD specs and right-trim their SKU string with 0x00 or other non-ASCII char (especially 0xFF like Corsair).

Can you change this to: `while (sku_len && (sku[sku_len - 1] <= 0x20 || sku[sku_len - 1] > 0x7F))`

Reply:
That's correct. ;)

Reply:
Do you want me to add a comment to make it clear?